### PR TITLE
[CHORE]: Convert Discover to Typescript + cleanup discover context

### PR DESCRIPTION
**Pull request by walmat**
## What changed (plus any additional context for devs)
Converts discover screen content to use typescript. Fixes up a lot of the redundancy with discover screen context too

## Screen recordings / screenshots

https://github.com/user-attachments/assets/68c9d16b-7801-4572-9c93-4bc02341b0ac

## What to test
Discover screen functionality
- searching
- clearing
- scroll to top
- etc etc

### CODE REVIEW

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/discover-typescript/805ede8a268eef8cd56d12d13441e3eb1467c85b.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/discover-typescript/805ede8a268eef8cd56d12d13441e3eb1467c85b.ipa) for 805ede8a268eef8cd56d12d13441e3eb1467c85b

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/discover-typescript/b5efb76cf657441ba06420ddc53df9351974763c.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/discover-typescript/b5efb76cf657441ba06420ddc53df9351974763c.ipa) for b5efb76cf657441ba06420ddc53df9351974763c

File: src/hooks/useOnAvatarPress.ts [hunk 1]
```diff
@@ -224,7 +224,7 @@ export default ({ screenType = 'transaction' }: UseOnAvatarPressProps = {}) => {
             actionKey: 'removePhoto',
             actionTitle: lang.t('profiles.profile_avatar.remove_photo'),
           },
-    ].filter(Boolean),
+    ].filter(Boolean) as readonly MenuActionConfig[],
```
Comment (greg-schrammel) [inline src/hooks/useOnAvatarPress.ts:227]: is this still needed? should be fixed with that lint fix pr

File: src/screens/discover/components/DiscoverSearchInput.tsx [hunk 2]
```diff
@@ -164,10 +184,19 @@ const ExchangeSearch = (
     };
   });
 
+  const searchInputValue = useAnimatedProps(() => {
+    // Removing the value when the input is focused allows the input to be reset to the correct value on blur
```
Comment (greg-schrammel) [inline src/screens/discover/components/DiscoverSearchInput.tsx:188]: I don't understand this comment, doesn't seem related to the rest?

**Review by greg-schrammel — APPROVED** [review]

**Review by benisgold — APPROVED** [review]

File: src/screens/discover/components/DiscoverSearchInput.tsx (hunk 2 above)
Comment (walmat) [reply src/screens/discover/components/DiscoverSearchInput.tsx:188]: It's similar to what we do for DappBrowser and Swaps search which let's us use an uncontrolled input in a way that feels controlled.

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/discover-typescript/413140871af36f0d947f8dc3932e25193cfb953d.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/discover-typescript/413140871af36f0d947f8dc3932e25193cfb953d.ipa) for 413140871af36f0d947f8dc3932e25193cfb953d

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/discover-typescript/810d67e658a912b1237aa8fdcbfaa697e9b3aed9.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/discover-typescript/810d67e658a912b1237aa8fdcbfaa697e9b3aed9.ipa) for 810d67e658a912b1237aa8fdcbfaa697e9b3aed9
